x-pack/docs/en/security/es-security-principles.asciidoc

```diff
@@ -0,0 +1,55 @@
+[[es-security-principles]]
+== {es} security principles
+Protecting your {es} cluster and the data it contains is of utmost importance.
+There are numerous security practices and considerations for keeping your
+cluster safe, such as not saving passwords in plain text files or making sure
+that users have appropriate permissions (most users shouldn't be a superuser).
+These guidelines are important, but the following information provides something
+greater. These principles are foundational to running {es} in a secure manner.
+
+[discrete]
+[[security-dont-run-without]]
+=== Run {es} with security enabled
+Never run an {es} cluster without security enabled. This principle cannot be
+overstated. Running {es} without security leaves your cluster exposed to anyone
+on the internet, permitting anyone to download, modify, or delete any data in
+your cluster.
+<<configuring-stack-security,Start the {stack} with security enabled>> or
+<<manually-configure-security,manually configure security>> to prevent
+unauthorized access to your clusters and ensure that internode communication
+is secure. 
+
+[discrete]
+[[security-dont-expose]]
+=== Protect {es} from public internet traffic
+Even with security enabled, never expose {es} to public internet traffic.
+Using an application to sanitize requests to {es} still poses risks, such as
+a malicious user writing <<search,`_search`>> requests that could overwhelm an
+{es} cluster and bring it down. If you need to connect devices from the internet
+to {es}, use some combination of a proxy, firewall, and VPN. The best
+combination of these technologies depends on your security needs and risks.
+
+[discrete]
+[[security-dont-expose-to-users]]
+=== Implement a barrier between {es} and users
+While you absolutely shouldn't expose {es} directly to the internet, you also
+shouldn't expose {es} directly to users. The following configurations provide
+different methods of user interaction with {es}:
+
+* *Best*: Use an application to make requests on behalf of users.
+* *Better*: If the previous configuration isn't possible, use an application to
+sanitize requests from users.
+* *Good*: If either of the previous configurations isn't possible, implement
+some mechanism to track user behaviors, such as requests that are submitted to
+specific nodes in the cluster.
+
+[discrete]
+[[security-dont-run-as-root]]
+=== Create appropriate users to run {es}
+Never run {es} as the `root` user. Running as the `root` user provides an avenue
+to circumvent the other security layers and permit a user to do *anything* on
+your server. {es}  will refuse to start if it detects that it is running as
+`root`, but it's worth double (or triple) checking the user that {es} runs as. 
```

Comment:
@bytebilly, I agree that the messaging is strange. This language is from four years ago, and I'm curious if this mention is even worth including in the docs anymore. I'll cc @nik9000 for some history, since he was the original author of this text back in PR [#23930](https://github.com/elastic/elasticsearch/commit/7fad7c675d73a5386d3bdea475c85c083dd68f07#diff-2d6718763cf5a7d5b98beefc23796acd26b8c5c096051ae70a06a4b2f0530b4bR12-R17).

Reply:
Partly we had it in the docs to explain why we added the check.

I agree this doesn't really fit with the conversation about roles and privileges but I don't think it sends a bad message. Its important not to run as root. Very very very important. So important you should probably double check when you first do the deployment that nothing got crossed up. Paranoia is good here. We'll check. We test that we check on every PR. But you should be paranoid here. Its important.

This language comes from a time before we even had a docker distribution and when a lot of folks were decompressing ES and just running it. Our rpm, deb, and docker distributions all create the user automatically. You should double check them because its important, but you shouldn't have to take any more action.

Reply:
I still feel that the message is too focused on not running as root, rather than to run on an unprivileged user.
What about this more focused and simplified message? (some language fix too)

> Never try to run Elasticsearch as the root user, as this would make any defense strategy not effective and permits a malicious user to do anything on your server.
>
> You must create a dedicated unprivileged user to run Elasticsearch. By default, the rpm, deb, docker, and Windows packages come with an elasticsearch user ready to use for this scope.